ca/ca.go

```diff
@@ -659,8 +663,70 @@ func (ca *CertificateAuthorityImpl) generateOCSPAndStoreCertificate(
 		// changes here, you should make sure they are reflected in orphan-finder.
 		ca.log.AuditErrf("Failed RPC to store at SA, orphaning certificate: serial=[%s] cert=[%s] err=[%v], regID=[%d], orderID=[%d]",
 			core.SerialToString(serialBigInt), hex.EncodeToString(certDER), err, regID, orderID)
+		if ca.orphanQueue != nil {
+			ca.queueOrphan(&orphanedCert{
+				DER:      certDER,
+				OCSPResp: ocspResp,
+				RegID:    regID,
+			})
+		}
 		return core.Certificate{}, err
 	}
 
 	return core.Certificate{DER: certDER}, nil
 }
+
+type orphanedCert struct {
+	DER      []byte
+	OCSPResp []byte
+	RegID    int64
```

Comment:
Where would this be used? AddCertificate doesn't use the order ID.

Reply:
It's used in logging IIRC. It's there, why not pipe it through? I don't think there is any other convenient way to match an orphaned certificate to the V1 or V2 API otherwise.

Reply:
It's logged in `generateOCSPAndStoreCertificate` already and `queueOrphan` and `integrateOrphan` don't log anything about the orphan themselves.

Reply:
k